### PR TITLE
Add test to trigger 'return value ignored' on Preconditions.checkNotNull

### PR DESCRIPTION
**Pull request by ewirch**
- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

See issue #578.

### CODE REVIEW

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/PreconditionsCheckNotNullCanIgnoreReturnValueTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import static org.junit.Assume.assumeThat;
+
+public class PreconditionsCheckNotNullCanIgnoreReturnValueTest {
+    @Rule
+    public SpotBugsRule spotbugs = new SpotBugsRule();
+
+    @Before
+    public void setUp() {
+        String property = System.getProperty("AUX_CLASSPATH");
+        assumeThat(property, notNullValue());
```
Comment (ThrawnCA) [inline spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/PreconditionsCheckNotNullCanIgnoreReturnValueTest.java]: Could just use `Assume.assumeNotNull`

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/PreconditionsCheckNotNullCanIgnoreReturnValueTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        for (String path : property.split(",")) {
+            spotbugs.addAuxClasspathEntry(Paths.get(path));
+        }
+    }
+
+    @Test
+    public void testDoNotWarnOnCanIgnoreReturnValue() {
+        BugCollection bugCollection = spotbugs.performAnalysis(Paths.get("../spotbugsTestCases/build/classes/java/main/bugPatterns/RV_RETURN_VALUE_IGNORED_Guava_Preconditions.class"));
```
Comment (ThrawnCA) [inline spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/PreconditionsCheckNotNullCanIgnoreReturnValueTest.java]: :+1: This is nice. I'm not a big fan of JUnit rules, but if they make SpotBugs unit testing this easy, I'll get over it.

**Comment by ThrawnCA** [comment]
@KengoTODA Looks like it's not fixed.

**Comment by KengoTODA** [comment]
@ThrawnCA yes my fix 139eefa hasn't been merged yet. I'll make another PR that contains his test and my fix.

**Pull request closed without merging**